webrecorder/browsertrix
# Improvements to collection creation and editing flow

Resolves https://github.com/webrecorder/browsertrix-cloud/issues/1333



### Changes

- Moves "Select Crawls" / "Select Uploads" steps into a single "Select Archived Items" dialog
- Refactors new collection metadata dialog to accept editing existing collection
- Prevents RWP component from rendering if there are no archived items (@Shrinks99 made a comment about this figma, but this prevents unnecessary requests when there isn't an archive to replay)
- Shows collection description at bottom of detail page at all times (@Shrinks99 seems useful to see even on archived items view?)
- Switches collection detail primary action to "Add Archived Items" if none are included (cc @Shrinks99)
- Displays friendlier "name taken" error
- Removes unused Collection edit route
- Upgrades markdown dependencies for fixes/improvements to description editing

### Manual testing

1. Log in and go to Collections
2. Click "New Collection" and complete creation. Verify you're redirected to the collections "Archived Items" list
3. Click "Add Archived Items" and update selection. Verify selecting crawls and updates works the same as it did in the previous iteration of the editor (further updates to follow in https://github.com/webrecorder/browsertrix-cloud/issues/1406)
4. Save selection. Verify detail page updates as expected
5. Click "Actions" -> "Edit Metadata" and update metadata. Verify metadata saves as expected
6. Click pencil icon next to "Description". Verify metadata dialog opens
7. Go back to "Collections" list
8. Click "..." -> "Edit Metadata" for a collection and update. Verify metadata saves as expected

### Screenshots

| Page | Image/video |
| ---- | ----------- |
| Collections | <img width="1034" alt="Screenshot 2023-12-05 at 7 55 44 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/604ae904-d28f-4088-bb24-562660394c23"> |
| Collections | <img width="761" alt="Screenshot 2023-12-05 at 8 10 17 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/4461eba1-fb8d-46c2-a3f9-3c23b6343327"> |
| Collection Detail | <img width="1126" alt="Screenshot 2023-12-05 at 8 10 28 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/7432db6e-d3c0-4492-a40d-6e19800360d4"> |
| Collection Detail | <img width="353" alt="Screenshot 2023-12-05 at 8 11 09 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/458ebb97-3e05-4cd9-b87e-a85912dbb0ec"> |
| Collection Detail | <img width="1145" alt="Screenshot 2023-12-05 at 8 10 35 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/99189e94-1a4d-462d-a8c9-8eb80a634743"> |
| Collection Detail | <img width="1147" alt="Screenshot 2023-12-05 at 8 10 49 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/5d69bc91-2d5e-4c64-bb9f-2389bed7c395"> |
| Collection Detail | <img width="428" alt="Screenshot 2023-12-05 at 7 46 08 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/df212e4a-b370-4c92-9f91-3d2604bd303c"> |
